### PR TITLE
Switch to intra-doc links in `core::hash`

### PR DESCRIPTION
**Pull request by camelid**
Part of #75080.

@rustbot modify labels: A-intra-doc-links T-doc T-rustdoc

### CODE REVIEW

File: library/core/src/hash/mod.rs [hunk 1]
```diff
@@ -503,10 +491,7 @@ pub trait BuildHasher {
 /// let hash_map = HashMap::<u32, u32, MyBuildHasher>::default();
 /// ```
 ///
-/// [`BuildHasher`]: trait.BuildHasher.html
-/// [`Default`]: ../default/trait.Default.html
-/// [method.default]: #method.default
-/// [`Hasher`]: trait.Hasher.html
+/// [method.default]: BuildHasherDefault::default
```
Comment (camelid) [inline library/core/src/hash/mod.rs:494]: This is failing. Should I try to resolve it as `Default::default` instead or just leave the original link?

Comment (jyn514) [reply library/core/src/hash/mod.rs:494]: If you wait a week until the beta branch, it will be fixed by #74489.

Comment (camelid) [reply library/core/src/hash/mod.rs:494]: Hmm, doesn't the beta branch happen today? https://rust-lang.zulipchat.com/#narrow/stream/122651-general/topic/When.20is.20the.20cutoff.20for.20the.20beta.20release.3F

Comment (jyn514) [reply library/core/src/hash/mod.rs:494]: Not until tomorrow: https://forge.rust-lang.org/release/process.html#master-bootstrap-update-t-1-day-wednesday

Comment (camelid) [reply library/core/src/hash/mod.rs:494]: Thanks! I love this line:

> Bask in your success.

**Comment by camelid** [comment]
Did that to retrigger the build.

**Comment by camelid** [comment]
Rebased so that it uses the latest beta.

**Pull request merged**